hyperspy/models/model1d.py

```diff
@@ -463,7 +462,7 @@ def set_signal_range(self, x1=None, x2=None):
 
         """
         try:
-            x1, x2 = signal_range_from_roi(x1)
+            x1, x2 = x1
         except TypeError:
```

Comment:
I am not sure about it: I think that if the length of the tuple doesn't match we want to raise the exception because the only valid inputs are a matching ROI or a number.

Reply:
Yes, I agree. Actually, it looks to me that it is not possible to get a `TypeError` with/without this PR? If so, shall we removed it?

Reply:
I think that it is fine as is: you can get a ``TypeError`` by passing a number as argument.

Reply:
Yes, indeed!